fitbenchmarking/core/tests/test_fitting_benchmarking_benchmark.py

```diff
@@ -0,0 +1,173 @@
+        dict_test(expected_unselected_minimzers, unselected_minimzers)
+        dict_test(expected_minimzers, self.options.minimizers)
+
+    @mock.patch('{}.loop_over_benchmark_problems'.format(FITTING_DIR))
+    def test_check_no_unselected_minimizers(self,
+                                            loop_over_benchmark_problems):
+        """
+        Checks benchmarking runs with no unselected minimizers
+        """
+        names = ["random_1", "random_3", "random_2"]
+        expected_names = sorted(names)
+        results = []
+        for name in names:
+            result_args = {'options': self.options,
+                           'problem': self.problem,
+                           'jac': self.problem.jac,
+                           'initial_params': self.problem.starting_values[0],
+                           'params': [],
+                           'chi_sq': 1,
+                           'name': name}
+            list_results = [fitbm_result.FittingResult(**result_args)
+                            for j in range(self.scipy_len)]
+            results.extend(list_results)
+        problem_fails = []
+        expected_unselected_minimzers = {"scipy": []}
+        expected_minimzers = copy.copy(self.all_minimzers)
+
+        loop_over_benchmark_problems.return_value = \
+            (results, problem_fails, expected_unselected_minimzers)
+        self.shared_tests(expected_names, expected_unselected_minimzers,
+                          expected_minimzers)
+
+    @mock.patch('{}.loop_over_benchmark_problems'.format(FITTING_DIR))
+    def test_check_unselected_minimizers(self, loop_over_benchmark_problems):
+        """
+        Checks benchmarking runs with a few unselected minimizers
+        """
+        names = ["random_1", "random_3", "random_2"]
+        expected_names = sorted(names)
+        results = []
+        for name in names:
+            result_args = {'options': self.options,
+                           'problem': self.problem,
+                           'jac': self.problem.jac,
+                           'initial_params': self.problem.starting_values[0],
+                           'params': [],
+                           'chi_sq': 1,
+                           'name': name}
+            list_results = [fitbm_result.FittingResult(**result_args)
+                            for j in range(self.scipy_len)]
+            results.extend(list_results)
+        problem_fails = []
+        expected_unselected_minimzers = {"scipy": ['SLSQP', 'Powell', 'CG']}
+        expected_minimzers = copy.copy(self.all_minimzers)
+        for keys, minimzers in expected_unselected_minimzers.items():
+            expected_minimzers[keys] = \
+                list(set(expected_minimzers[keys]) - set(minimzers))
+
+        loop_over_benchmark_problems.return_value = \
+            (results, problem_fails, expected_unselected_minimzers)
+        self.shared_tests(expected_names, expected_unselected_minimzers,
```

Comment:
I currently struggle to see why this shared test in this case will work. Since to me currently the mocked `loop_over_benchmark_problems.return_value` returns exactly the same as that of the previous test. What am I missing?

Reply:
This test checks to see if the ``unselected_minimzers`` are set correctly.